src/openshift/cluster.ts

```diff
@@ -66,17 +66,19 @@ export class Cluster extends OpenShiftItem {
         Cli.getInstance().showOutputChannel();
     }
 
-    static async openshiftConsole(context: OpenShiftObject): Promise<void> {
-        if (context) {
-            open(`${context.getName()}/console`);
+    static async openshiftConsole(): Promise<void> {
+        let consoleUrl: string;
+        const versionInfo = await Cluster.odo.execute(Command.clusterVersionDescribe(), process.cwd(), false);
+        if (versionInfo.error === null) {
+            const routeObj = await Cluster.odo.execute(Command.getOpenshiftClusterRoute());
```

Comment:
What is output for OpenShift 3.x for this command?

Reply:
For 3.x, there is no namespace `openshift-console` and hence the route does not work here. 

For 4.x, the cluster url is generates using operator and we get the route using the namespace for `openshift-console`.

Reply:
then clusterVersionDescribe is not needed,  just analyse getOpenshiftClusterRoute result.

Reply:
But `getOpenshiftClusterRoute` for 3.x does not throw an error. It responds with empty items list. Do you want to just use the condition over the `items` object and iterate ?

Reply:
@dgolovin After conversation with openshift dev team, they too suggested that `oc get clusterversion` is the correct differentiator as it works only on 4.x.